com.unity.ml-agents.extensions/Runtime/Match3/Match3Sensor.cs

```diff
@@ -88,22 +91,43 @@ public Match3Sensor(AbstractBoard board, GridValueProvider gvp, int oneHotSize,
         /// <returns></returns>
         public static Match3Sensor CellTypeSensor(AbstractBoard board, Match3ObservationType obsType, string name)
         {
-            return new Match3Sensor(board, board.GetCellType, board.NumCellTypes, obsType, name);
+            var maxBoardSize = board.GetMaxBoardSize();
+            return new Match3Sensor(board, board.GetCellType, maxBoardSize.NumCellTypes, obsType, name);
         }
 
         /// <summary>
-        /// Create a sensor that encodes the cell special types as observations.
+        /// Create a sensor that encodes the cell special types as observations. Returns null if the board's
+        /// NumSpecialTypes is 0 (indicating the sensor isn't needed).
         /// </summary>
         /// <param name="board">The abstract board.</param>
         /// <param name="obsType">Whether to produce vector or visual observations</param>
         /// <param name="name">Name of the sensor.</param>
         /// <returns></returns>
         public static Match3Sensor SpecialTypeSensor(AbstractBoard board, Match3ObservationType obsType, string name)
         {
-            var specialSize = board.NumSpecialTypes == 0 ? 0 : board.NumSpecialTypes + 1;
+            var maxBoardSize = board.GetMaxBoardSize();
+            if (maxBoardSize.NumSpecialTypes == 0)
+            {
+                return null;
+            }
+            var specialSize = maxBoardSize.NumSpecialTypes + 1;
             return new Match3Sensor(board, board.GetSpecialType, specialSize, obsType, name);
         }
 
+        [Conditional("DEBUG")]
+        void CheckBoardSizes()
```

Comment:
static method for this somewhere?

Reply:
Since it's in two places.  Not a big deal either way.

Reply:
Moved to AbstractBoard.